File: SERVIDOR SDR_SCR/lib/presentation/ucb/ucbpresenter/ucb_get_attendance_control_list_presenter.py
import asyncio
from lib.domain.boundary.output.get_attendance_control_list_presenter import GetAttendanceControlListPresenter
from lib.presentation.ucb.ucbviewmodel.ucb_get_attendance_control_list_view_model import UCBGetAttendanceControlListViewModel
from lib.presentation.ucb.ucbview.ucb_get_attendance_control_list_view import UCBGetAttendanceControlListView
from lib.domain.dto.response.get_attendance_control_list_response import GetAttendanceControlListResponse

class UCBGetAttendanceControlListPresenter(GetAttendanceControlListPresenter):
    def __init__(self, view: UCBGetAttendanceControlListView):
        self.__view = view

    async def displayAttendanceControlList(self, response: GetAttendanceControlListResponse):
        model = UCBGetAttendanceControlListViewModel(response.getAttendanceControlLists())
        await self.__view.sendStudentLists(model)

File: SERVIDOR CENTRAL/lib/presentation/ucb/ucbpresenter/ucb_get_attendance_control_list_presenter.go
package ucbpresenter

import (
	"gitlab.com/jorgechavezrnd/servidorseminariogo2/lib/domain/dto/response"
	"gitlab.com/jorgechavezrnd/servidorseminariogo2/lib/presentation/ucb/ucbview"
	"gitlab.com/jorgechavezrnd/servidorseminariogo2/lib/presentation/ucb/ucbviewmodel"
)

// UCBGetAttendanceControlListPresenter ...
type UCBGetAttendanceControlListPresenter struct {
	View ucbview.UCBGetAttendanceControlListView
}

// DisplayAttendanceControlList ...
func (ucbGetAttendanceControlListPresenter UCBGetAttendanceControlListPresenter) DisplayAttendanceControlList(response response.GetAttendanceControlListResponse) {
	model := ucbviewmodel.UCBGetAttendanceControlListViewModel{
		StudentsPresentSlice:                   response.StudentsPresentSlice,
		StudentsMissingSlice:                   response.StudentsMissingSlice,
		NumberOfDoubtfulStudents:               response.NumberOfDoubtfulStudents,
		PictureWithFacesDetectedAndNamesBase64: response.PictureWithFacesDetectedAndNamesBase64,
	}

	ucbGetAttendanceControlListPresenter.View.SendAttendanceControlList(model)
}
